notebooks/stitches_dev/functions/gridded_recipe.R

```diff
@@ -0,0 +1,203 @@
+                        archive_variable = x["archive_variable"][[1]], 
+                        archive_model = x["archive_model"][[1]], 
+                        archive_ensemble = x["archive_ensemble"][[1]], 
+                        stitching_id = x["stitching_id"][[1]], 
+                        archive_start_yr = as.integer(x["archive_start_yr"[[1]]]), 
+                        archive_end_yr = as.integer(x["archive_end_yr"][[1]]), stringsAsFactors = FALSE)
+      
+      return(out)
+      
+      
+    }
+    
+  }) %>% 
+    dplyr::bind_rows() -> 
+    out 
+  
+  return(out)
+  
+}
+
+
+
+# Go through a recipe and ensure that all of the periods have the same archive 
+# and target period length, if not update to reflect the target period length. 
+# Otherwise you'll end up with extra years in the stitched data. This is really 
+# only an issue for the final period of target data because sometimes that period is somewhat short. 
+# Args 
+# rp: a data frame of the recipe 
+# Return: a recipe data frame that has target and archive periods of the same length.
+handle_final_period <- function(rp){
+  
+  out <- data.frame(stringsAsFactors = FALSE)
+  
+  # For each row in rp check to see if the length 
+  # of the target and the archive years are the same. 
+  for(row in 1:nrow(rp)){
+    
+    len_target <- rp$target_end_yr[row] - rp$target_start_yr[row]
+    len_archive <- rp$archive_end_yr[row] - rp$archive_start_yr[row]
+    
+    if(len_target == len_archive){
+      
+      out <- rbind(out, rp[row, ])
+      
+    } else if (len_target < len_archive) {
+      
+      # Figure out how much shorter the target period is than the archive period. 
+      diff <- len_archive - len_target
+      updated <- rp[row, ]
+      updated$archive_end_yr <- updated$archive_end_yr - 1
+      
+      # Add the updated row to the data frame. 
+      out <- rbind(out, updated)
+    } else {
+      
+      stop("don't know how to handle this problem!")
+      
+    }
+  }
+  
+  return(out)
+  
+}
+
+
+
+
+# Generate recipe for the stitched gridded product 
+# (based on the notebook 5 code)
+# Args 
+#   target_data: a data frame of the target data 
+#   archive_data: the data frame of the archive data to match on
+#   n: int number of permutations based on the neighborhood matching TODO write some test to account for when tol is small 
+#   tol: default set to 0.1, it defines the threshold for the neighborhood matching process
+# Return: data frame containing the following columns, stitching_id, target_start_yr, target_end_yr, 
+#     archive_start_yr, archive_end_yr, and file. This is all of the information that will be needed 
+#     by python to stitch the gridded product. 
+generate_gridded_recipe <- function(target_data, archive_data, n, tol = 0.1){
+  
+  # Load the complete archive. 
+  complete_archive <- read.csv("inputs/main_raw_pasted_tgav_anomaly_all_pangeo_list_models.csv", 
```

Comment:
This isn't actually used in this code. And also I don't think we would want to hard code in the archive like this in general, although it is pretty moot right now since the R code is temporary. 

So I think maybe comment out this block

Reply:
Agreed I don't love hard coding these things and we will want a smart way load the archive of data and I am guessing that this will be stored as a pickle or something.